graphs/possible_bipartition.py

```diff
@@ -5,8 +5,43 @@ def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: Generally Linear O(N), but more specifically 
+            O(N+E) or O(V+E), in which E is edges & V for vertices,
+        Space Complexity: We're initializing 3 new data structures ->
+            performance depends on the length of these sets, Linear O(N)
     """
-    pass
+    # guard-clause for an early-exit
+    if not dislikes:
+        return True
 
+    queue = deque()
+    visited = set()
+    red = set()
+    green = set()
+
+    if dislikes[0]:
+        start_node = 0
+    else:
+        start_node = 1
```

Comment:
🤔 Consider a case where the graph has multiple nodes disconnected and they are all indexed at the beginning of the adjacency list `dislikes`. For example:
`dislikes = [ [], [], [], [4, 5], [5,1], [3]]`. How might you refactor your code to account for this edge case?